### PR TITLE
Use Find-modules from Hunter

### PR DESCRIPTION
**Pull request by ruslo**

### CODE REVIEW

File: ogles_gpgpu/CMakeLists.txt [hunk 1]
```diff
@@ -56,50 +61,36 @@ else()
...
-    if(NOT OSMESA_LIBRARY)
-      message(FATAL_ERROR "OGLES_GPGPU was configured to use OSMesa, but the library could not be found")
-    endif()
 
-    get_filename_component(OSMESA_PATH ${OSMESA_LIBRARY} DIRECTORY)
+    find_package(osmesa REQUIRED)
+    find_package(glapi REQUIRED)
+    find_package(egl REQUIRED)
 
-    find_library(GLAPI_LIBRARY glapi HINTS ${OSMESA_PATH} PATHS ${OSMESA_PATH})
```
Comment (headupinclouds) [inline ogles_gpgpu/CMakeLists.txt:67]: We need to make sure find_package() replacements use the `${OSMESA_PATH}` hint.  I believe these are both part of the osmesa installation.

Comment (ruslo) [reply ogles_gpgpu/CMakeLists.txt:67]: It is possible that there are two versions of `glapi.so` in system?

Comment (headupinclouds) [reply ogles_gpgpu/CMakeLists.txt:67]: Actually, I think `glapi.so` is specific to `osmesa`.  I believe there is an `EGL` that is specific to `osmesa`, however.

Comment (headupinclouds) [reply ogles_gpgpu/CMakeLists.txt:67]: Installation paths on trusty (cloud instance):
```
/usr//lib/x86_64-linux-gnu/mesa-egl/
├── ld.so.conf
├── libEGL.so -> libEGL.so.1.0.0
├── libEGL.so.1 -> libEGL.so.1.0.0
├── libEGL.so.1.0.0
├── libGLESv2.so -> libGLESv2.so.2.0.0
├── libGLESv2.so.2 -> libGLESv2.so.2.0.0
├── libGLESv2.so.2.0.0
├── libOpenVG.so.1 -> libOpenVG.so.1.0.0
└── libOpenVG.so.1.0.0
```

```
/usr/lib/x86_64-linux-gnu/libglapi.so.0.0.0
/usr/lib/x86_64-linux-gnu/libglapi.so.0
/usr/lib/x86_64-linux-gnu/libglapi.so
```

```
/usr/lib/x86_64-linux-gnu/libOSMesa.so
/usr/lib/x86_64-linux-gnu/libOSMesa.so.6
/usr/lib/x86_64-linux-gnu/libOSMesa.so.6.5.3
```

```
/usr/lib/x86_64-linux-gnu/pkgconfig/osmesa.pc
```

**Pull request merged**